tests/model_indexes/tests.py

```diff
@@ -0,0 +1,41 @@
+from django.db import models
+from django.test import TestCase
+
+from .models import Book
+
+
+class IndexesTests(TestCase):
+
+    def test_raises_error_without_field(self):
+        msg = "At least one field is required to define an index."
+        with self.assertRaisesMessage(ValueError, msg):
+            models.Index()
+
+    def test_max_name_length(self):
+        # When user provides index name longer than 30 characters
+        msg = "Index names cannot be longer than 30 characters."
+        with self.assertRaisesMessage(ValueError, msg):
+            models.Index(fields=['title'], name='looooooooooooong_index_name_idx')
+
+        # To check if auto-generated name is not more than 30 characters
+        index = models.Index(fields=['title'])
+        index.model = Book
+        self.assertTrue(len(index.name) <= 30)
```

Comment:
Since index name generation is deterministic, I don't think this is needed. `test_name_auto_generation` already checks for a specific name. If this index is truncated is some different way, then it would be okay to keep it but check for the actual name rather than testing the length.